### ISSUE TITLE
Cannot connect on linux

### ISSUE THREAD

**Issue by e-wartal** [FreeJoy-Team/FreeJoy#100]
[30058.419960] usb usb1-port2: attempt power cycle
[30059.071876] usb 1-2: new full-speed USB device number 35 using xhci_hcd
[30059.072142] usb 1-2: Device not responding to setup address.
[30059.280137] usb 1-2: Device not responding to setup address.
[30059.487799] usb 1-2: device not accepting address 35, error -71
[30059.615752] usb 1-2: new full-speed USB device number 36 using xhci_hcd
[30059.615851] usb 1-2: Device not responding to setup address.
[30059.823981] usb 1-2: Device not responding to setup address.
[30060.031809] usb 1-2: device not accepting address 36, error -71
[30060.031937] usb usb1-port2: unable to enumerate USB device

**Comment by Reksotiv** [comment]
Did you copy 99-hid-FreeJoy.rules file to /etc/udev/rules.d from linux release archive?

**Comment by e-wartal** [comment]
Thanks for quick response, I have just copy that file, but still doesn't work, and I have the same issue, I have tried with different USB cable length and different socket and i have the same problem all the time, quite strange. There is also problem on windows "Device Descriptor request Failed", maybe there is problem with cheap "hardware" ...

**Comment by Reksotiv** [comment]
If device does not work in Windows, then the problem is hardware. Have you tried reflashing firmware?

**Comment by vostrenkov** [comment]
https://github.com/FreeJoy-Team/FreeJoyWiki/issues/19

**Comment by e-wartal** [comment]
I have 3 boards and none of them seems to be work, I have reflashed firmware but still have the same problem.

**Comment by e-wartal** [comment]
Hardware: HW-267

**Comment by vostrenkov** [comment]
> Hardware: HW-267

So what about R10 value?
http://amitesh-singh.github.io/stm32/2017/05/27/Overcoming-wrong-pullup-in-blue-pill.html

**Comment by e-wartal** [comment]
I changed R10 to 1.5K and still have the same problem.

**Comment by vostrenkov** [comment]
Here is prebuilt ST's example project from official USB library. Are you able to run it?
[JoyStickMouse.zip](https://github.com/FreeJoy-Team/FreeJoy/files/6055201/JoyStickMouse.zip)

**Comment by e-wartal** [comment]
Yeah, still the same issue, I tested in this same way before but without result, this is huge issue I see there are a lots of articles and youtube video about it, It looks like hardware issue. This board has also poor quality of soldering

**Comment by pipes80-collab** [comment]
My arceli has a 251 resistor smd, I don't know the value

**Comment by reinholdPL** [comment]
had the same problem. then erased and flashed blue pill under windows 10 according to https://github.com/FreeJoy-Team/FreeJoyWiki/blob/master/eng/Flashing-firmware-with-ST-Link-programmer.md
blue pill started showing properly under linux as freejoy.
